src/python/pants/backend/scala/bsp/rules.py

```diff
@@ -141,14 +141,22 @@ async def handle_bsp_scalac_options_request(
     jvm: JvmSubsystem,
 ) -> HandleScalacOptionsResult:
     wrapped_target = await Get(WrappedTarget, AddressInput, request.bsp_target_id.address_input)
-    resolve = wrapped_target.target[JvmResolveField].normalized_value(jvm)
+    coarsened_targets = await Get(CoarsenedTargets, Addresses([wrapped_target.target.address]))
+    assert len(coarsened_targets) == 1
+    coarsened_target = coarsened_targets[0]
+    resolve = await Get(CoursierResolveKey, CoarsenedTargets([coarsened_target]))
+
+    # TODO: This duplicates code in the scalac rules. Find a way to merge!
+    output_file = f"{coarsened_target.representative.address.path_safe_spec}.scalac.jar"
```

Comment:
Calling a static method (and leaving the TODO there) would be good.

Reply:
Will do.

Reply:
Done.